### PR TITLE
Introduce helper for creating hermetic iframes and images in tests.

### LINKED ISSUE #3100: amp-youtube flakes
```
Chromium 45.0.2454 (Ubuntu 0.0.0) amp-youtube renders FAILED
	Error: timeout of 5000ms exceeded. Ensure the done() callback is being called in this test.
Chromium 45.0.2454 (Ubuntu 0.0.0): Executed 1955 of 1983 (1 FAILED) (skipped 20) (0 secs / 42.33 secs)
```

### PR DESCRIPTION
**Pull request by cramforce**
And use them to make YT and Facebook tests less flaky.

Fixes #3100

### CODE REVIEW

File: testing/iframe.js [hunk 1]
```diff
@@ -317,6 +317,53 @@ export function expectBodyToBecomeVisible(win) {
...
 /**
+ * For the given iframe, makes the creation of iframes and images
+ * create elements that do not actually load their underlying
+ * resources.
+ * Calling `triggerLoad` makes the respective resource appear loaded.
+ * Calling `triggerError` on the respective resources makes them
+ * appear in error state.
+ * @param {!Window} win
+ */
+export function doNotLoadExternalResourcesInTest(win) {
```
Comment (erwinmombay) [inline testing/iframe.js]: good stuff

File: extensions/amp-facebook/0.1/test/test-amp-facebook.js [hunk 2]
```diff
@@ -26,8 +29,11 @@ describe('amp-facebook', function() {
   const fbPostHref = 'https://www.facebook.com/zuck/posts/10102593740125791';
   const fbVideoHref = 'https://www.facebook.com/zuck/videos/10102509264909801/';
 
-  function getFBPost(href, opt_embedAs) {
+  function getFBPost(href, opt_embedAs, opt_noFakeResources) {
     return createIframePromise().then(iframe => {
+      if (!opt_noFakeResources) {
```
Comment (jridgewell) [inline extensions/amp-facebook/0.1/test/test-amp-facebook.js]: Double negative! 😛

File: extensions/amp-facebook/0.1/test/test-amp-facebook.js [hunk 3]
```diff
@@ -73,27 +81,28 @@ describe('amp-facebook', function() {
...
-        iframe.onload = function() {
-          iframe.contentWindow.postMessage({
-            sentinel: 'amp-test',
-            type: 'requestHeight',
-            is3p: true,
-            height: 666,
-          }, '*');
-        };
+    return getFBPost(fbPostHref, undefined,
+        /* opt_noFakeResources */ true).then(ampFB => {
```
Comment (mkhatib) [inline extensions/amp-facebook/0.1/test/test-amp-facebook.js]: Why fake resources wouldn't work for this one?

File: testing/iframe.js [hunk 4]
```diff
@@ -317,6 +317,53 @@ export function expectBodyToBecomeVisible(win) {
...
+  const createElement = win.document.createElement;
+  win.document.createElement = function(tagName) {
+    const element = createElement.apply(this, arguments);
+    tagName = tagName.toLowerCase();
+    if (tagName == 'iframe' || tagName == 'img') {
```
Comment (mkhatib) [inline testing/iframe.js]: Might be worth adding video and audio as well?

**Comment by mkhatib** [comment]
Awesome stuff! Would love to move instagram, twitter and all the other embed types (we've a lot of these) to using this. Mind either updating the PR or filing a tracking issue to update the tests to use this?

File: testing/iframe.js (hunk 4 above)
Comment (mkhatib) [inline testing/iframe.js]: Though if our tests use local video and audio files it might be less critical.

**Comment by mkhatib** [comment]
LGTM. Please file issues to track adding video/audio to fake load, also an issue to track upgrading other tests to use this. And lint errors.

File: extensions/amp-facebook/0.1/test/test-amp-facebook.js (hunk 2 above)
Comment (cramforce) [inline extensions/amp-facebook/0.1/test/test-amp-facebook.js]: Yeah, I hated myself. But better to make a flag for the special case.

File: extensions/amp-facebook/0.1/test/test-amp-facebook.js (hunk 3 above)
Comment (cramforce) [inline extensions/amp-facebook/0.1/test/test-amp-facebook.js]: It makes its own fake (fixture) resource that actually does something.

File: testing/iframe.js (hunk 4 above)
Comment (cramforce) [inline testing/iframe.js]: Yeah, if we have reasonable local files there isn't as big of a deal. Video and audio support the `source` tags which makes things more complicated.

**Comment by cramforce** [comment]
Really weird, my local lint is happy.

**Comment by erwinmombay** [comment]
what were the lint errors?

**Comment by mkhatib** [comment]
@erwinmombay I believe they were indentation errors.

**Pull request merged**